cocos/animation/core/transform-array.ts

```diff
@@ -0,0 +1,227 @@
+import { Vec3 } from '../../core/math/vec3';
+import { Quat } from '../../core/math/quat';
+import { Transform } from './transform';
+
+const TRANSFORM_STRIDE_IN_FLOATS = 10;
+
+const TRANSFORM_STRIDE_IN_BYTES = Float64Array.BYTES_PER_ELEMENT * TRANSFORM_STRIDE_IN_FLOATS;
+
+const ROTATION_OFFSET = 3;
+
+const SCALE_OFFSET = ROTATION_OFFSET + 4;
+
+/**
+ * Array-buffer-based transform array.
+ */
+export class TransformArray {
+    public static get BYTES_PER_ELEMENT () {
+        return TRANSFORM_STRIDE_IN_BYTES;
+    }
+
+    constructor (length?: number);
+
+    constructor (buffer: ArrayBuffer, byteOffset?: number, length?: number);
+
+    constructor (bufferOrLength?: number | ArrayBuffer, byteOffset?: number, length_?: number) {
+        if (typeof bufferOrLength === 'undefined') {
+            this._data = new Float64Array();
```

Comment:
Is `Float64Array` necessary? It seems wasted

Reply:
float64 can be utmostly compatible with `number`, even I'm not sure if it's neccesary.

Reply:
float32 precision should be enough, here you need to save a little memory, you can refer to the precision used by other engines.